File: app/src/main/java/com/paradoxo/amadeus/util/voz/EscutadaoraService.java
package com.paradoxo.amadeus.util.voz;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;

import androidx.annotation.Nullable;

public class EscutadaoraService extends Service {
    VozParaTexto vozParaTexto;

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        vozParaTexto = new VozParaTexto(getApplicationContext());
        vozParaTexto.backgroundVoiceListener.run();

        return super.onStartCommand(intent, flags, startId);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        vozParaTexto.backgroundVoiceListener.interrupt();
    }


}

File: app/src/main/java/com/paradoxo/amadeus/util/voz/EscutadaoraService.kt
package com.paradoxo.amadeus.util.voz

import android.app.Service
import android.content.Intent
import android.os.IBinder

class EscutadaoraService : Service() {
    private lateinit var vozParaTexto: VozParaTexto

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        vozParaTexto = VozParaTexto(applicationContext)
        vozParaTexto.backgroundVoiceListener.run()
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onBind(intent: Intent): IBinder? = null

    override fun onDestroy() {
        super.onDestroy()
        vozParaTexto.backgroundVoiceListener.interrupt()
    }
}
